### ISSUE TITLE
Time interval between picture changes

### ISSUE THREAD

**Issue by sks7g** [Caleb9/syno-photo-frame#6]
Switching between the images takes a long time, I haven't stopped it, but a bit faster would be nice.

I'm currently using WLAN, LAN would also be possible. 
I don't know exactly, are the images transferred from the NAS in full size? 
I can't configure the transfer size in the album (Synology), does the API offer this?

But maybe there is still a delay in syno-photo-frame. 
Is there a fixed time interval between the individual image changes that can be set (parameter, config file)? 
If so, it would be nice if this could be changed externally (argument, config).
BR
Erik

**Comment by Caleb9** [comment]
### TL;DR

> Is there a fixed time interval between the individual image changes that can be set (parameter, config file)?

Yes, the `--interval` or `-i` option accepts amount of time between each photo in seconds. The default value is 30 seconds, so set it to something smaller (but not less than 5 seconds - there's still some time it takes to download and process the next photo). For example:

`syno-photo-frame https://{YOUR SHARING LINK} --interval 20`

Use the `--help` option to see all the available switches.

### Long version

> I don't know exactly, are the images transferred from the NAS in full size?

Depends what you mean. When you add any photo to Synology Photos, behind the scenes there's always a number of JPEG "thumbnails" of different sizes generated from the original photo. That's how it supports raw photos (e.g. CR2 Canon format). It stores your original files, but the API responds with JPEGs displayed by the client (whether it's a browser or syno-photo-frame). There is a parameter in the API that allows control over which size you want to fetch, but currently syno-photo-frame does not provide a way to control this, and fetches the maximum size image for best quality.

Still, if I understand you correctly, what you need is already available by the `--interval` setting mentioned above. :)

HTH

**Comment by fsr5525** [comment]
syno-photo-frame works with Synology’s [Sharing Link].
http://xxx:5000/mo/sharing/lw5pf5XM9

Doesn't it work as a [folder link] like the link below?
http://xxx:5000?launchApp=SYNO.Foto.AppInstance#/shared_space/folder/9932
